component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/service/IconResolver.java

```diff
@@ -69,6 +71,14 @@ protected void init() {
                         .stream()
                         .filter(it -> !it.endsWith(".svg"))
                         .collect(toList());
+        log.info("[IconResolver] SVG supported: {}, patterns: {}.", isSupportsSvg(), patterns);
+    }
+
+    private Boolean studioSupportsSvgOrTrue() {
+        if (System.getProperty("talend.studio.version") != null) {
```

Comment:
We need to keep the current behavior (ie handling only png) for compatibility. Also, we need to provide in some cases the use of svg (studio modernization).
Since the beginning, the framework has always made an intensive use of system properties for many things including some specific platform detection (beam, recordfactorybldr, studio, etc.)